pyccel/codegen/printing/ccode.py

```diff
@@ -250,18 +254,38 @@ def _print_Complex(self, expr):
                         PyccelMul(expr.imag, ImaginaryUnit()))))
 
     def _print_PythonComplex(self, expr):
-        self._additional_imports.add("complex.h")
+        self._additional_imports.add("complex")
         return self._print(PyccelAssociativeParenthesis(PyccelAdd(expr.real_part,
                         PyccelMul(expr.imag_part, ImaginaryUnit()))))
 
     def _print_ImaginaryUnit(self, expr):
         return '_Complex_I'
 
+    def _print_ModuleHeader(self, expr):
+        name = expr.module.name
+        # TODO: Add classes and interfaces
+        funcs = '\n\n'.join('{};'.format(self.function_signature(f)) for f in expr.module.funcs)
+
+        # Print imports last to be sure that all additional_imports have been collected
+        imports = [*expr.module.imports, *map(Import, self._additional_imports)]
```

Comment:
Maybe this can be a separate issue.
this code in python
```python
from pyccel.decorators import types
@types('int,int')
def rec(b,y):
    return b + y
z = rec(2,4)
```
generated in c `prog_pro.c`
```c
#include <pro.h>
#include <stdlib.h>
int main()
{
    long z;

    z = rec(2,4);
    return 0;
}
```
and `pro.h`
```c
#include <stdlib.h>

/*........................................*/
long rec(long b, long y)
{
    long Out_0001;
    Out_0001 = b + y;
    return Out_0001;
}
/*........................................*/
```
here we don't need `<stdlib.h>` but it is imported in the two files, so we need to remove it until time needed.
and for the local header file imported we need to use the C convention for including local  `#include "pro.h"` instead of `#include <pro.h>`  (it doesn't lead to error in compiling but it is better to use C standard coding style).

Reply:
>here we don't need `<stdlib.h>`

Which file doesn't need `stdlib.h`? `prog_pro.c`, `pro.h`, or `pro.c`? The second file that you have copied above is `pro.c` not `pro.h`. 
As a first attempt I have left all the same imports in `pro.h` as in `pro.c` as there is no simple way to know which ones are needed for the header and which are needed for the body of the function.

> we need to use the C convention for including local headers `#include "pro.h"` instead of `#include <pro.h>`

I agree but pyccel doesn't save any information in the `Import` class apart from the name. So currently at the codegen stage we don't know if the header is local or global. I think we should open a new issue for this as it may require modifications in several places, especially if we can import user-provided fortran files to the python (I think this was possible at one point, but I have no idea if it still works, @saidctb ?). If the user provides a file then we don't know if it is local or global.
Unless you have an idea for a quick fix?

Reply:
> Which file doesn't need `stdlib.h`? `prog_pro.c`, `pro.h`, or `pro.c`? The second file that you have copied above is `pro.c` not `pro.h`.
> As a first attempt I have left all the same imports in `pro.h` as in `pro.c` as there is no simple way to know which ones are needed for the header and which are needed for the body of the function.

the header file `.h` will include all necessary libraries and macros, for the file with the functions declaration need only to include the header (which is the header itself contains necessary includes).

Reply:
Ah! I see what you mean. I will do that

Reply:
done

Reply:
great Emily. but we still need have to include the `_additional_imports`  in the `moduleHeader`.
to see that try to use a math function.

Reply:
Oops you are right, I fixed it

Reply:
I have created an issue for the `""` vs `<>` problem. Are you happy to resolve this conversation now?

Reply:
hhh, we went back to the same point.
I see that we need to get the `_additional_imports`  from `.c` file printer to `.h` printer, which currently I am no sure if we can do that.

Reply:
of course, I am happy with your work.
you did well. these are just structuring stuff.

Reply:
If the imports aren't needed in the header then I think it's better if they're only included in the `.c` file. That way a minimal number of additional things are imported into other files. That will help to avoid name collisions. I don't know if there's any way of avoiding a double import of something like `complex.h` right now though. I will open an issue

Reply:
I see what you mean. for C style all includes, structures, and macros are in the `.h` file.
the `.c` files will include just the header file.

> I don't know if there's any way of avoiding a double import of something like complex.h right now though. I will open an issue

If you revise the header file you will see that it is wrapped with `ifndef` ... `endif`.
so there will be no problem if we import that header many times.
and the code will be well organized.
in the future, we may add only one function in a file and create one header for all functions within these files.
what do you think.

Reply:
This is really a question of convention. In my coding classes I was taught to limit the number of includes in the header so something isn't included unless it needs to be / isn't included without you being aware.
E.g.
I write a file my printer.h which contains an implementation of a function `printf` that I want to use in the rest of my code for some reason (e.g to always write to stderr, I don't have a good example in mind, but I'm sure there is one).
I then import `printer.h` in my file where I also import `BigClass.h`, `BigClass.h` has a debug method which uses `printf` from stdio. If `stdio.h` was only imported in `BigClass.c` where it is needed then this will not cause a problem, but if it is imported in `BigClass.h` to limit the number of imports in `BigClass.c` then my `main.c` will contain 2 implementations of `printf` which will be a problem

> in the future, we may add only one function in a file and create one header for all functions within these files.
what do you think.

1 function per `.c` file and 1 `.h` file for all `.c` files? That is not standard practice in c I can't see any advantages to doing it this way. Did you have any in mind? It makes name clashes more likely and you have to be more careful with the order in which you construct your `.h` file so any structures containing structures are declared in the right order